transmogrifier/sources/xml/dspace_dim.py

```diff
@@ -259,6 +221,73 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
 
         return fields
 
+    @classmethod
+    def get_contents(cls, xml: Tag) -> list[str]:
+        return [
+            str(contents.string)
+            for contents in xml.find_all(
+                "dim:field",
+                element="description",
+                qualifier="tableofcontents",
+                string=True,
+            )
+        ]
+
+    @classmethod
+    def get_dates(cls, xml: Tag, source_record_id: str) -> list[timdex.Date]:
+        dates = []
+        for date_element in xml.find_all("dim:field", element="date", string=True):
+            date_value = str(date_element.string.strip())
+            if validate_date(date_value, source_record_id):
+                if date_element.get("qualifier") == "issued":
+                    date_object = timdex.Date(value=date_value, kind="Publication date")
+                else:
+                    date_object = timdex.Date(
+                        value=date_value, kind=date_element.get("qualifier") or None
+                    )
+                dates.append(date_object)
+        dates.extend(cls._get_coverage_dates(xml, source_record_id))
+        return dates
+
+    @classmethod
+    def _get_coverage_dates(cls, xml: Tag, source_record_id: str) -> list[timdex.Date]:
+        coverage_dates = []
+        for coverage_value in [
+            str(coverage_element.string)
+            for coverage_element in xml.find_all(
+                "dim:field", element="coverage", qualifier="temporal", string=True
+            )
+        ]:
+            if "/" in coverage_value:
+                date_object = cls._parse_date_range(coverage_value, source_record_id)
+            else:
+                date_object = timdex.Date(note=coverage_value, kind="coverage")
+            if date_object:
+                coverage_dates.append(date_object)
+        return coverage_dates
+
+    @classmethod
+    def _parse_date_range(
+        cls, coverage_value: Tag, source_record_id: str
+    ) -> timdex.Date | None:
+        """Parse date range value and return a Date object if it is validated."""
+        split = coverage_value.index("/")
+        gte_date = coverage_value[:split]
+        lte_date = coverage_value[split + 1 :]
+        if validate_date_range(
```

Comment:
A good point and worthy of discussion, but this does feel like a slightly different type of validation than what we're doing init'ing a `TimdexRecord` which is more checking types rather than the value's content. I'll add it to the agenda!